File: concept/17_recursion/stairs_count/stairs_count.dart
import 'dart:io';

int stair(int s) {
  if (s < 0) {
    return 0;
  }
  if (s == 0) {
    return 1;
  }

  int res = stair(s - 1) + stair(s - 2);

  return res;
}

void main() {
  print("Enter n : ");
  int n = int.parse(stdin.readLineSync()!);

  int ways = stair(n);
  print("N is : $n  result is $ways ");
}

File: concept/17_recursion/stairs_count/stairs_count.cpp
#include <iostream>
using namespace std;

int stair(int s)
{
    if (s < 0)
    {
        return 0;
    }
    if (s == 0)
    {
        return 1;
    }

    int res = stair(s - 1) + stair(s - 2);

    return res;
}
int main()
{
    cout << "Enter n : ";
    int n;
    cin >> n;

    int ways = stair(n);
    cout << "N is : " << n << " result is " << ways << endl;
}